ambry-commons/src/main/java/com.github.ambry.commons/ByteBufferReadableStreamChannel.java

```diff
@@ -30,16 +36,25 @@
 public class ByteBufferReadableStreamChannel implements ReadableStreamChannel {
   private final AtomicBoolean channelOpen = new AtomicBoolean(true);
   private final AtomicBoolean channelEmptied = new AtomicBoolean(false);
-  private final ByteBuffer buffer;
+  private final List<ByteBuffer> buffers;
   private final int size;
 
   /**
    * Constructs a {@link ReadableStreamChannel} whose read operations return data from the provided {@code buffer}.
    * @param buffer the {@link ByteBuffer} that is used to retrieve data from on invocation of read operations.
    */
-  public ByteBufferReadableStreamChannel(ByteBuffer buffer) {
-    this.buffer = buffer;
-    size = buffer.remaining();
+  public ByteBufferReadableStreamChannel(ByteBuffer buffer) { this(Arrays.asList(buffer)); }
+
+  /**
+   * Constructs a {@link ReadableStreamChannel} whose read operations return data from the provided {@code bufferList}.
+   * @param bufferList list of {@link ByteBuffer} that is used to retrieve data from on invocation of read operations.
+   */
+  public ByteBufferReadableStreamChannel(List<ByteBuffer> bufferList) {
+    if (bufferList == null || bufferList.isEmpty()) {
+      throw new IllegalArgumentException("Input List of ByteBuffer cannot be null/empty for ByteBufferReadableStreamChannel");
+    }
+    buffers = new ArrayList<>(bufferList);
```

Comment:
If at all the`ByteBufferReadableStreamChannel` is constructed using an immutable list, then it would break at`bufferIterator.remove();` in `readInto`.